IdentityCore/src/throttling/model/MSIDThrottlingModel429.m

```diff
@@ -0,0 +1,121 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+
+#import <Foundation/Foundation.h>
+#import "MSIDThrottlingModel429.h"
+#import "NSError+MSIDThrottlingExtension.h"
+
+@implementation MSIDThrottlingModel429
+static NSInteger const Default429Throttling = 60;
+static NSInteger const MaxRetryAfter = 3600;
+
+- (instancetype)initWithRequest:(id<MSIDThumbprintCalculatable>)request
+                    cacheRecord:(MSIDThrottlingCacheRecord *)cacheRecord
+                  errorResponse:(NSError *)errorResponse
+                    accessGroup:(NSString *)accessGroup
+{
+    self = [super initWithRequest:request cacheRecord:cacheRecord errorResponse:errorResponse accessGroup:accessGroup];
+    if (self)
+    {
+        self.thumbprintType = MSIDThrottlingThumbprintTypeStrict;
+        self.thumbprintValue = [request strictRequestThumbprint];
+        self.throttleDuration = Default429Throttling;
+    }
+    return self;
+}
+
+
+/**
+ 429 throttle conditions:
+ - HTTP Response code is 429 or in 5xx range
+ - OR Retry-After in response header
+ */
++ (BOOL)isApplicableForTheThrottleModel:(NSError *)errorResponse
+{
+    /**
+     In SSO-Ext flow, it can be both MSAL or MSID Error. If it's MSALErrorDomain, we need to extract information we need (error code and user info)
+     */
+    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, nil, @"Throttling: error response domain: %@", errorResponse.domain);
+
+    BOOL res = NO;
+    
+    NSString *httpResponseCode = [errorResponse msidGetUserInfoValueWithMSIDKey:MSIDHTTPResponseCodeKey orMSALKey:@"MSALHTTPResponseCodeKey"];
```

Comment:
Just a note: I would like @jasoncoolmax and @antrix1989 to cross-reference this one and make sure they are fine with bringing MSAL into the common core context.